pkg/scheduler/framework/plugins/noderesources/requested_to_capacity_ratio.go

```diff
@@ -45,32 +45,12 @@ func NewRequestedToCapacityRatio(plArgs runtime.Object, handle framework.Handle)
 		return nil, err
 	}
 
-	shape := make([]helper.FunctionShapePoint, 0, len(args.Shape))
-	for _, point := range args.Shape {
-		shape = append(shape, helper.FunctionShapePoint{
-			Utilization: int64(point.Utilization),
-			// MaxCustomPriorityScore may diverge from the max score used in the scheduler and defined by MaxNodeScore,
-			// therefore we need to scale the score returned by requested to capacity ratio to the score range
-			// used by the scheduler.
-			Score: int64(point.Score) * (framework.MaxNodeScore / config.MaxCustomPriorityScore),
-		})
-	}
-
-	resourceToWeightMap := make(resourceToWeightMap)
-	for _, resource := range args.Resources {
-		resourceToWeightMap[v1.ResourceName(resource.Name)] = resource.Weight
-		if resource.Weight == 0 {
-			// Apply the default weight.
-			resourceToWeightMap[v1.ResourceName(resource.Name)] = 1
-		}
```

Comment:
Since this plugin can still be used via v1beta1, we need to maintain this defaulting logic, but in 
https://github.com/kubernetes/kubernetes/blob/5c586201186b1f31d16257511baef7793112ae76/pkg/scheduler/apis/config/v1beta1/defaults.go#L195 as follows:

```
func SetDefaults_RequestedToCapacityRatioArgs(obj *v1beta1.RequestedToCapacityRatioArgs) {	
  if len(obj.Resources) == 0 {
      // If no resources specified, used the default set.		
     obj.Resources = append(obj.Resources, defaultResourceSpec...)	
  }
  for i := range obj.ScoringStrategy.Resources { 
    if obj.ScoringStrategy.Resources[i].Weight == 0 { 
       obj.ScoringStrategy.Resources[i].Weight = 1 
     } 
  }
}
```

Reply:
FYI: #101946 is also touching this, so the latter-merged PR may expect a conflict.